backend/inspirehep/accounts/api.py

```diff
@@ -66,3 +75,32 @@ def get_current_user_orcid():
 
 def get_current_user_remote_orcid_account():
     return RemoteAccount.query.filter_by(user_id=current_user.get_id()).one_or_none()
+
+
+def check_permissions_for_private_collections(
+    roles=[], collections=None, readwrite=False
+):
+    if collections:
+        private_collections = set(collections) - set(
+            current_app.config["NON_PRIVATE_LITERATURE_COLLECTIONS"]
+        )
+        for col in private_collections:
+            if readwrite:
+                roles.append(col + "-read-write")
+            else:
+                roles.append(col + "-read")
+
+    # superuser can do everything
+    roles.append(Roles.superuser.value)
+    user_roles = {role.name for role in current_user.roles}
+
+    # read-write roles to have read access
+    if not readwrite:
+        user_roles = {
+            role if not role.endswith("-read-write") else role[:-6]
+            for role in user_roles
+        }
```

Comment:
I am not sure if I understand it. Why can't you add `-read` for everything in line 91 (`read-write` has always access to read)?